main.go

```diff
@@ -12,6 +24,125 @@ func init() {
 	_ = flag.Set("logtostderr", "true")
 }
 
+func GetCurrentNode(clientset kubernetes.Clientset) (*corev1.Node, error) {
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		return nil, errors.New("Environment variable NODE_NAME is not set")
+	}
+
+	node, err := clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node: %w", err)
+	}
+
+	return node, nil
+}
+
+func UpdateNodeLabels(
+	clientset *kubernetes.Clientset,
+	instanceData *metadata.InstanceData,
+) error {
+	if instanceData == nil {
+		return fmt.Errorf("instance data received from Linode metadata service is nil")
+	}
+
+	node, err := GetCurrentNode(*clientset)
+	if err != nil {
+		return fmt.Errorf("failed to get the node: %w", err)
+	}
+
+	klog.Infof("Updating node labels with Linode instance data: %v", instanceData)
+
```

Comment:
missing the tags labels too? going to be hard to get a `[]string` into a label but i think we should make an attempt in this PR

Reply:
make a csv string?

Reply:
oof i hate it... no one can target pods to `decorator.linode.com/tags=tag1,tag2,tag3` especially when they're all over the place and out of order? i literally have no better ideas tho

Reply:
seems like the only good option here is to request upstream that tags be a kv pair. enumeration would be ok too, but I think given the structure of tags inside Linode that might be more problematic elsewhere. 

without that I think we're probably hosed for lke scheduling based on tags.

Reply:
@luthermonson perhaps we could sort before we add tags? its super annoying but would at least get you consistent ordering..

Reply:
What about if we only sync `key=value` tags?

So for the tags "foo", "bar", "baz=bat" we would only use the "baz=bat" tag and we'd label the node as such?

Reply:
i like this but let's go another level on the "subdomain", `tags.decorator.linode.com/baz=bat`, this avoids any clashing of the existing decorator labels. if the tag is just `baz` then set the value to something expected

`baz=bat` becomes `tags.decorator.linode.com/baz=bat`
`baz` becomes `tags.decorator.linode.com/baz=default`

Reply:
Ok I like that too, the chunky question is then what should the default value be? We can just have an empty string "" as the value

baz becomes `tags.decorator.linode.com/baz=""`

Reply:
Yea, luther and I discussed that yesterday. Empty string is fine for a default value and you can split on `=` or `:`

```
tags: ["foo:bat", "bar=", "baz"]

output labels:
tags.decorator.linode.com/foo: "bat"
tags.decorator.linode.com/bar: ""
tags.decorator.linode.com/baz: ""
```

Reply:
For multiple operators (`:`, or `=`) scenarios, I think we should only respect the first operator:
```
tags: ["foo:bar:baz", "bar=foo=baz", "baz=foo:bar"]

output labels:
tags.decorator.linode.com/foo: "bar:baz"
tags.decorator.linode.com/bar: "foo=baz"
tags.decorator.linode.com/baz: "foo:bar"
```

And for the duplicated keys scenarios, I think we can always use the latter value: 
```
tags: ["foo=bar", "foo=baz"]

output labels:
tags.decorator.linode.com/foo: "baz"
```

Do you guys think this is a good idea?

Reply:
Both of those seem reasonable to me ^

Reply:
I agree with only honoring the first operator, that makes sense to me. Would be interested in other opinions though.

For the duplicated keys I don't think we can honor both/either correctly. AFAIK labels are implemented as a `map[string]string` so we'd have to either honor the first one or the last one. Since we don't have any enforced ordering I think we're going to fall into an area of undefined behavior. I'm thinking we want to report an error, not apply either, and try to propagate it up through the observability stack.

Maybe there's a better way of notifying the user that there's a duplicate?

Reply:
I think we can push validation of the map to APINext. I image it can validate that there are no duplicates in the map.

Reply:
@hcwagner Ah sorry, I forgot labels were in a map. Thanks for reminding me! Yep, in that case, we can override it with a latter value?

```
tags: ["foo=bar", "foo=baz"]

output labels:
tags.decorator.linode.com/foo: "baz"
```

Reply:
@zliang-akamai You're probably right, I suppose the tags are basically `[]string` so you can have multiple instances of the same key with different values. 

Imo what we should do is a stable sort of the tags we get from the API and then set the K:V pairs incrementally. That way we've at least got consistent behaviour.

Reply:
Seems like it's a decent starting point. I still suggest we'll want to provide some visibility into whether there are duplicates at some point. Even though it will consistently end with the same result I definitely see the potential for weirdness occurring in the scheduler when pods are assigned based on affinities set against labels that the decorator flips every time it comes through. It wouldn't be immediately obvious what the problem is with the scheduling if you're debugging from k9s or something like that unless you directly see the flipping of the label value occur.

Reply:
Thanks everyone! The tags handling logic has been implemented. Let me know if you have any feedback/suggestion about it, ^_^